src/core/sgrubberband.cpp

```diff
@@ -85,25 +87,41 @@ SGRubberband::SGRubberband( const QVector<QgsPoint> &points, Qgis::GeometryType
   }
 }
 
-QSGGeometryNode *SGRubberband::createLineGeometry( const QVector<QgsPoint> &points, float width )
+QSGGeometryNode *SGRubberband::createLineGeometry( const QVector<QgsPoint> &points )
 {
+  // outline node contains the regular node that the outline node is rendered behind the regular node
   QSGGeometryNode *node = new QSGGeometryNode;
+  QSGGeometryNode *outlineNode = new QSGGeometryNode;
   QSGGeometry *sgGeom = new QSGGeometry( QSGGeometry::defaultAttributes_Point2D(), points.count() );
+  QSGGeometry *outlineSGGeom = new QSGGeometry( QSGGeometry::defaultAttributes_Point2D(), points.count() );
   QSGGeometry::Point2D *vertices = sgGeom->vertexDataAsPoint2D();
+  QSGGeometry::Point2D *outlineVertices = outlineSGGeom->vertexDataAsPoint2D();
 
   int i = 0;
   for ( const QgsPoint &pt : points )
   {
-    vertices[i++].set( static_cast<float>( pt.x() ), static_cast<float>( pt.y() ) );
+    outlineVertices[i].set( static_cast<float>( pt.x() ), static_cast<float>( pt.y() ) );
+    vertices[i].set( static_cast<float>( pt.x() ), static_cast<float>( pt.y() ) );
+    i += 1;
   }
 
-  sgGeom->setLineWidth( static_cast<float>( width ) );
+  sgGeom->setLineWidth( static_cast<float>( mWidth ) );
   sgGeom->setDrawingMode( QSGGeometry::DrawLineStrip );
   node->setGeometry( sgGeom );
   node->setMaterial( &mMaterial );
   node->setFlag( QSGNode::OwnsGeometry );
   node->setFlag( QSGNode::OwnedByParent );
-  return node;
+
+  outlineSGGeom->setLineWidth( static_cast<float>( mWidth * 3 ) );
```

Comment:
I think here I'd go for mWidth + 4 (to ever only add a 2 pixel buffer outside of the main width), I think the outline shouldn't grow in width alongside the main width.

Reply:
Actually sorry, we'll need to send the outline width separately from the rubber band class as we need the device pixel ratio to have a static outline buffer width.

Reply:
It's nice when the first comment raises a question and the second one answers it. :)

Reply:
Initially did it like you said with extra parameter, then I thought it might not be needed to make the API slightly simpler and only add it on demand. Well, the demand came :)

Reply:
![image](https://github.com/opengisch/QField/assets/2820439/f9266fbb-6f3b-4ebd-9103-56e89e22a9b2)
@nirvn pushed the requested version